src/Image.zig

```diff
@@ -42,17 +44,49 @@ pub const Stream = io.StreamSource;
 
 pub const EncoderOptions = AllImageFormats.ImageEncoderOptions;
 
-pub const SaveInfo = struct {
-    width: usize,
-    height: usize,
-    encoder_options: EncoderOptions,
+pub const AnimationLoopInfinite = -1;
+
+pub const AnimationFrame = struct {
+    pixels: PixelStorage,
+    duration: f32,
+};
+
+pub const Animation = struct {
+    frames: std.ArrayListUnmanaged(AnimationFrame),
+    loop_count: i32 = AnimationLoopInfinite,
```

Comment:
If we decide to add support for APNG one day just having this data will not allow for correct replay of the animation. Should we at least have some idea how will we handle that issue? I guess we could just add additional fields that will get some default values for GIF animations.

Reply:
I just read the APNG specification, it has the same ability as GIF to be able to only store the changed contents per frame using (x,y, width, height),clear method and an extra blend control. 

For now, I will let the decoder decode the full image for each frame and the future encoders for GIF and APNG could try to be smart and find optimizations for each frame like doing a bounding box of visible pixels and only store those in the frame data. If you want to keep those graphic control data, you'll need to use the format directly.

Reply:
Ah, cool. I didn't realize GIF had those features as well.